### DISCUSSION TITLE
Alguém pode me ajudar por favor?

### DISCUSSION THREAD

**Discussion by alissondias99** [frontendbr/forum#2217] [Perguntas e Respostas] [answered]
Estou estudando Java Script juntamente com HTML e CSS, e quando estava praticando com uns exercícios eu travei nesse aqui.
Tudo que ele tem que fazer é pegar um preço antigo e um novo, e verificar se teve um aumento ou queda, mas por algum motivo sempre que eu digito um número acima de 100 ele desconsidera o else if e aparece uma mensagem parecida com essa, "O produto custava R$100 e agora custa R$99.
Hoje o produto está mais caro.
O aumento foi de R$-1".
Se alguém puder dar uma olhada no código e me dar umas dicas eu agradeceria muito.

<!DOCTYPE html>
<html lang="pt-br">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Preço mudou</title>
    <style>
        body{
            font-size: 14pt;
        }
        button{
            font-size: 15pt;
            padding: 1em;
        }
    </style>
</head>
<body>
    <h1>Preço mudou</h1>

    <button onclick="calcular()">Verificar</button>

    <section id="calculo">
        Clique no botão para verificar a mudança no preço
    </section>

    <script>

        function calcular() {

            var preco = window.prompt('Digite um preço antigo do produto')
            
            var npreco = window.prompt('Digite o novo preço do produto')

            var calculo = document.getElementById('calculo')

            if (preco < npreco) {

                let diferenca =  npreco - preco

                calculo.innerHTML = `
                O produto custava R$${preco} e agora custa R$${npreco}.<br/>
                Hoje o produto está mais caro. <br/>
                O aumento foi de R$${diferenca} <br/>
                `
            } 
            
            else if (preco > npreco) {

                let diferenca = preco - npreco 

                calculo.innerHTML = `
                O produto custava R$${preco} e agora custa R$${npreco}. <br/>
                Hoje o produto está mais barato. <br/>`
        }
    }
    </script>
</body>
</html>

**Comment by lipegusmao** [comment, marked as answer]
Opa! Beleza?

Só precisava dar um Number( ) (ou parseFloat) nas variáveis preco e npreco, pra identificar como número real.

```
<script>

    function calcular() {

        var preco = Number(window.prompt('Digite um preço antigo do produto'))
        
        var npreco = Number(window.prompt('Digite o novo preço do produto'))

        var calculo = document.getElementById('calculo')

        if (preco < npreco) {

            let diferenca =  npreco - preco

            calculo.innerHTML = `
            O produto custava R$${preco} e agora custa R$${npreco}.<br/>
            Hoje o produto está mais caro. <br/>
            O aumento foi de R$${diferenca} <br/>
            `
        } 
        
        else if (preco > npreco) {

            let diferenca = preco - npreco 

            calculo.innerHTML = `
            O produto custava R$${preco} e agora custa R$${npreco}. <br/>
            Hoje o produto está mais barato. <br/>`
    }
}
</script>
```

**Reply by alissondias99** [reply]
Muito obrigado, me ajudo muito, e agora eu sei como lidar com esse erro em exercícios futuros, vlw D+++

**Reply by lipegusmao** [reply]
Disponha! Bons estudos! 🤜🏼🤛🏼
